src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -0,0 +1,378 @@
+      div({ style: { display: 'flex' } }, [
+        div(
+          {
+            style: {
+              backgroundColor: colors.accent(0.2),
+              color: colors.dark(),
+              padding: '0.5rem',
+              borderRadius: '2rem',
+              fontSize: 14,
+              height: 24,
+              width: 24,
+              display: 'flex',
+              alignItems: 'center',
+              justifyContent: 'center',
+              fontWeight: 600,
+            },
+          },
+          [number]
+        ),
+        div({ style: { marginLeft: 10, height: '3rem' } }, [
+          h3({ style: { marginTop: 0, marginBottom: '0.5rem' } }, [header]),
+          div({ style: { fontSize: 12 } }, [subheader]),
+        ]),
+      ]),
+      headerAction,
+    ]),
+    div(
+      {
+        style: {
+          backgroundColor: 'white',
+          border: `1px solid ${colors.dark(0.5)}`,
+          borderRadius: 10,
+          height: 300,
+          padding: '1rem',
+          marginTop: '0.5rem',
+        },
+      },
+      [
+        valuesSets && valuesSets.length > 0 && _.flatMap((valuesSet) => valuesSet.values, valuesSets).length > 0
+          ? div(
+              { style: { display: 'flex', flexDirection: 'column', height: '100%', overflowY: 'auto' } },
+              _.map(
+                ([i, valuesSet]) =>
+                  // valuesSet.values.length > 0 &&
+                  valuesSet.values &&
+                  valuesSet.values.length > 0 &&
+                  h(Fragment, [
+                    h(DatasetBuilderSelectorSubHeader, { key: `${valuesSet.header}-${i}` }, [valuesSet.header]),
+                    _.map(([j, value]) => {
+                      return div(
+                        {
+                          style: {
+                            display: 'flex',
+                            padding: '0.5rem',
+                            border: `1px solid ${colors.light()}`,
+                            width: '100%',
+                            marginTop: '0.3rem',
+                            fontSize: 13,
+                          },
+                          key: `${valuesSet.header}-${value.name}-${j}`,
+                        },
+                        [
+                          h(
+                            LabeledCheckbox,
+                            {
+                              key: `${valuesSet.header}-${value.name}-${j}-checkbox`,
+                              checked: _.flow(
+                                _.filter(
+                                  (selectedValuesSet: ValuesSet<T>) => selectedValuesSet.header === valuesSet.header
+                                ),
+                                _.flatMap((selectedValuesSet: ValuesSet<T>) => selectedValuesSet.values),
+                                _.map((selectedValue) => selectedValue.name),
+                                _.includes(value.name)
+                              )(selectedValuesSets),
```

Comment:
If this was java, I'd recommend putting this into a separate function so it has a name and its meaning is more obvious (e.g., `isChecked(valueSet, value selectedValueSets)`). Not sure if that makes sense in the UI code.